.github/workflows/check-dependencies.yaml

```
@@ -1,63 +1,24 @@
 name: Check Dependencies
 
-on: pull_request
+on:
+  - pull_request
+  - workflow_call
 
 jobs:
-  find:
-    runs-on: ubuntu-latest
-    outputs:
-      python: ${{ steps.find-python.outputs.result }}
-      web: ${{ steps.find-web.outputs.result }}
-    steps:
-      - uses: actions/checkout@v4
-      - run: |
-          curl https://raw.githubusercontent.com/SituDevelopment/automation/main/find-by-extension -o find-by-extension
-          chmod +x find-by-extension
-      - id: find-python
-        run: |
-          if [[ -n $(./find-by-extension ".py") ]]; then
-              echo "result=true" >> "$GITHUB_OUTPUT"
-          else
-              echo "result=false" >> "$GITHUB_OUTPUT"
-          fi
-      - id: find-web
-        run: |
-          if [[ -n $(./find-by-extension ".cjs" ".css" ".html" ".html.jinja" ".js" ".json" ".jsx" ".md" ".scss" ".ts" ".tsx" ".yaml" ".yml") ]]; then
-            echo "result=true" >> "$GITHUB_OUTPUT"
-          else
-            echo "result=false" >> "$GITHUB_OUTPUT"
-          fi
-
   pip:
-    if: fromJSON(needs.find.outputs.python)
-    needs: find
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v4
         with:
           python-version: "3.10"
-      - run: |
-          if [[ -f requirements.txt ]]; then
-              pip install -r requirements.txt
-          else
-              echo "requirements.txt not found"
-              exit 1
-          fi
+      - run: pip install -r requirements.txt
 
   npm:
-    if: fromJSON(needs.find.outputs.web)
-    needs: find
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
           node-version: "18"
-      - run: |
-          if [[ -f package.json && -f package-lock.json ]]; then
-              npm install
-          else
-              echo "package.json and/or package-lock.json not found"
-              exit 1
-          fi
+      - run: npm install
```

This should be using a clean install for as it is an automated test environment, docs are [here](https://docs.npmjs.com/cli/v8/commands/npm-ci)

```suggestion
      - run: npm clean-install
```